src/ripple/app/tests/ValidatorList_test.cpp

```diff
@@ -0,0 +1,228 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright 2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/basics/TestSuite.h>
+#include <ripple/app/misc/ValidatorList.h>
+
+namespace ripple {
+namespace tests {
+
+class ValidatorList_test : public ripple::TestSuite
+{
+private:
+    static
+    PublicKey
+    asPublicKey(RippleAddress const& raPublicKey)
+    {
+        auto const& blob = raPublicKey.getNodePublic();
+
+        if (blob.empty())
+            LogicError ("Can't convert invalid RippleAddress to PublicKey");
+
+        return PublicKey(Slice(blob.data(), blob.size()));
+    }
```

Comment:
This is an exact duplicate of `static asPublicKey` in ValidatorList.cpp. Why not expose that function and reuse it?